### PR TITLE
fix(perf): do less work loading config

### PR DESCRIPTION
**Pull request by wraithgar**
This adds some lazy loading, inlines some single use functions, and also
removes the "define" function in the definitions file.  That was
guarding against something that isn't worth the runtime to check for.

### CODE REVIEW

**Comment by wraithgar** [comment]
~Draft because this won't automatically run the npm tests. I will do that locally to make sure nothing broke there.~

npm tests fixed, added a config test to make sure we keep the definitions sorted (will help guard against the one thing `define()` was doing)

File: workspaces/config/lib/index.js [hunk 1]
```diff
@@ -695,6 +687,7 @@ class Config {
       }
 
       if (this.localPrefix && hasPackageJson) {
+        const rpj = require('read-package-json-fast')
```
Comment (wraithgar) [inline workspaces/config/lib/index.js:690]: We use none of the normalization here, we should consider using the underlying JSON parser in this situation.

File: workspaces/config/lib/type-defs.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 const nopt = require('nopt')
 
-const { Umask, validate: validateUmask } = require('./umask.js')
+const { validate: validateUmask } = require('./umask.js')
 
-const semver = require('semver')
+class Umask {}
+class Semver {}
```
Comment (wraithgar) [inline workspaces/config/lib/type-defs.js:6]: @H4ad the class is meaningless it's just a unique "object" to hang nopt definitions off of.  I opted for the same as `Umask` here instead. Thoughts?

Comment (H4ad) [reply workspaces/config/lib/type-defs.js:6]: If it doesn't matter, I think it's okay.

I'd rather have our own version than import from `semver`, even though the import time from `classes/semver` is being measured in `us`.

**Review by lukekarrys — APPROVED** [review]

**Comment by wraithgar** [comment]
Gonna add map-workspaces to this

**Pull request merged**